include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -43,54 +133,76 @@ struct DefaultKernelName;
 template <typename KernelName = DefaultKernelName>
 class device_policy
 {
+    static inline sycl::queue
+    __get_default_queue()
+    {
+        static sycl::queue __q(sycl::default_selector_v);
+        return __q;
+    }
+
   public:
     using kernel_name = KernelName;
 
     device_policy() = default;
     template <typename OtherName>
-    device_policy(const device_policy<OtherName>& other) : q(other.queue())
-    {
-    }
+    device_policy(const device_policy<OtherName>& other) : q(other.queue()) {}
     explicit device_policy(sycl::queue q_) : q(q_) {}
     explicit device_policy(sycl::device d_) : q(d_) {}
-    operator sycl::queue() const { return q; }
+
+    operator sycl::queue() const { return queue(); }
     sycl::queue
     queue() const
     {
-        return q;
+        return q.__get_queue();
     }
 
+#if _ONEDPL_PREDEFINED_POLICIES
+    explicit device_policy(__internal::__global_instance_tag __t) : q(__t, /*factory*/__get_default_queue) {}
+
+  protected:
+    device_policy(__internal::__global_instance_tag __t, __internal::_QueueFactory __f) : q(__t, __f) {}
+#endif
+
   private:
-    sycl::queue q;
+    static_assert(sizeof(__internal::__queue_holder) == sizeof(sycl::queue));
+    static_assert(alignof(__internal::__queue_holder) == alignof(sycl::queue));
+    __internal::__queue_holder q;
```

Comment:
I understand that it's potentially more changes than "really necessary" but how about renaming `q`. for example to `__holder`.. The reason why I am asking about that is `queue` public method implementation. `q.__get_queue` looks weird to me.

Reply:
No problem. Right, I did not want to change what does not have to be changed, but now you gave me an excuse :)